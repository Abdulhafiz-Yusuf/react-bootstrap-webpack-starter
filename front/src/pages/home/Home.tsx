import React from 'react';
// import {  } from 'react-router-dom';
import Jumbotron from 'reactstrap/lib/Jumbotron';
import { Link } from 'react-router-dom';
import HomeInfo from './styled/HomeInfo';
import MainTitle from './styled/MainTitle';
import LightNote from './styled/LightNote';
import FadeInEntrance from '../../components/fadeInEntrance';

type OwnProps = any;
type Props = OwnProps;

function Home() {
  return (
    <FadeInEntrance>
      <Jumbotron>
        <HomeInfo>
          <MainTitle>ReactJS 16.4+ Bootstrap 4</MainTitle>
          <h2>
            with Hot Reload (<i>react-hot-loader 4+</i>
            )!!!
          </h2>
          <h2>and React 16+ Context API</h2>
          <h2>and React Router v4</h2>
          <h2>and webpack 4.x</h2>
          <h2>and Typescript</h2>
          <h2>
            and styled-components (
            <LightNote>
              so keep using SCSS like styles and benefit performant css-in-js
            </LightNote>
            )
          </h2>
          <h1>Starter</h1>
          <p>
            <Link className="btn btn-success btn-lg" to={'/about'}>
              <i className="fa fa-info" />
              &nbsp; go to about
            </Link>
          </p>
        </HomeInfo>
      </Jumbotron>
    </FadeInEntrance>
  );
}

Home.displayName = 'Home';

export default Home;
