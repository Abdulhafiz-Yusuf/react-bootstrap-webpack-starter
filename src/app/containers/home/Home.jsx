import React      from 'react';
import Jumbotron  from '../../components/jumbotron/Jumbotron.jsx';
import classNames from 'classnames';
import { Link }   from 'react-router';

class Home extends React.Component {
  constructor(props) {
    super(props);
    this.init();
  }

  init() {
    this.state = {
      animated    : true,
      viewEnters  : false
    };
  }

  componentWillMount() {
    this.state = {
      viewEnters  : true
    };
  }

  processViewAnimationClasses() {
    const homeViewClasses = classNames({
      'animatedViews'    : this.state.animated,
      'view-enter'       : this.state.viewEnters
    });
    return homeViewClasses;
  }

  render() {
    return(
      <div
        key="homeView"
        className={this.processViewAnimationClasses()}>
        <Jumbotron>
          <h1>
            React + Bootstrap
          </h1>
          <p>
            starter
          </p>
          <p>
            react + react-router + classnames + bootstrap + react-bootstrap + animate.css + webpack
          </p>
          <p>
            <Link
              className="btn btn-success btn-lg"
              to={'/about'}>
              <i className="fa fa-info"></i>
              &nbsp;
              go to about
            </Link>
          </p>
        </Jumbotron>
      </div>
    );
  }
}

export default Home;
