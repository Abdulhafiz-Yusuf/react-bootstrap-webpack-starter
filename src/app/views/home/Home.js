// @flow weak

import React, {
  PureComponent
}                     from 'react';
import PropTypes      from 'prop-types';
import {Jumbotron}    from '../../components';
import cx             from 'classnames';
import { Link }       from 'react-router-dom';
import AnimatedView   from '../../components/animatedView/AnimatedView';

class Home extends PureComponent {
  static propTypes= {
    // react-router 4:
    match:    PropTypes.object.isRequired,
    location: PropTypes.object.isRequired,
    history:  PropTypes.object.isRequired
  };

  render() {
    return(
      <AnimatedView>
        <Jumbotron>
          <h1>
            ReactJS + Bootstrap
          </h1>
          <h2>
            with Hot Reload!!!
          </h2>
          <h2>
            and React Router v4
          </h2>
          <h2>
            and webpack 3.x
          </h2>
          <h1>
            Starter
          </h1>
          <p>
            <Link
              className="btn btn-success btn-lg"
              to={'/about'}>
              <i className="fa fa-info" />
              &nbsp;
              go to about
            </Link>
          </p>
        </Jumbotron>
      </AnimatedView>
    );
  }
}

export default Home;
