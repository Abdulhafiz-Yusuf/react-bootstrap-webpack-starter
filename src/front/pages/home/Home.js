// @flow

// #region imports
import React, { PureComponent } from 'react';
import {
  type Match,
  type Location,
  type RouterHistory,
} from 'react-router-dom';
import styled from 'styled-components';
import { Jumbotron } from '../../components';
import { Link } from 'react-router-dom';
// #endregion

// #region flow types
type Props = {
  // react-router 4:
  match: Match,
  location: Location,
  history: RouterHistory,

  ...any,
};
type State = any;
// #endregion

// #region styled-components
const HomeInfo = styled.div``;

const MainTitle = styled.h1`
  color: #222 !important;
  font-weight: 800;
`;

const LightNote = styled.i`
  font-size: 0.7em;
`;
// #endregion

class Home extends PureComponent<Props, State> {
  // #region lifecycle
  render() {
    return (
      <div>
        <Jumbotron>
          <HomeInfo>
            <MainTitle>ReactJS 16 + Bootstrap</MainTitle>
            <h2>
              with Hot Reload (<i>react-hot-loader 3.1+</i>)!!!
            </h2>
            <h2>and React Router v4</h2>
            <h2>and webpack 3.x</h2>
            <h2>
              and CSSModule (<LightNote>
                so keep using SCSS as you did before but import your class in
                your components like it were JS files
              </LightNote>)
            </h2>
            <h1>Starter</h1>
            <p>
              <Link className="btn btn-success btn-lg" to={'/about'}>
                <i className="fa fa-info" />
                &nbsp; go to about
              </Link>
            </p>
          </HomeInfo>
        </Jumbotron>
      </div>
    );
  }
  // #endregion
}

export default Home;
