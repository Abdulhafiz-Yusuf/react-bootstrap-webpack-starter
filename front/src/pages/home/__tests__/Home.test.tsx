import React from 'react';
import { MemoryRouter } from 'react-router';
import { render } from '@testing-library/react';
import '@testing-library/react/cleanup-after-each';
import Home from '../Home';

let rootElement: any = null;

describe('Home page', () => {
  beforeEach(() => {
    rootElement = document.createElement('div');
    document.body.appendChild(rootElement);
  });

  afterEach(() => {
    rootElement && document.body.removeChild(rootElement);
    rootElement = null;
  });

  it('renders as expected', () => {
    const { container } = render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>,
      rootElement,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
