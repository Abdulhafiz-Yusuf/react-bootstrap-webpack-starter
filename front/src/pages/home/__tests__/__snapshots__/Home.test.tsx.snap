// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders as expected 1`] = `
<div
  class="sc-bxivhb gbmved"
>
  <div
    class="jumbotron"
  >
    <div
      class="sc-bdVaJa bDWFJH"
    >
      <h1
        class="sc-bwzfXH jJLXBi"
      >
        ReactJS 16.11+ Bootstrap 4
      </h1>
      <h2>
        with Hot Reload (
        <i>
          react-hot-loader 4+
        </i>
        )!!!
      </h2>
      <h2>
        and React 16+ Context API
      </h2>
      <h2>
        and React Router v5
      </h2>
      <h2>
        and webpack 4.x
      </h2>
      <h2>
        and Typescript
      </h2>
      <h2>
        and styled-components (
        <i
          class="sc-htpNat jfNlAg"
        >
          so keep using SCSS like styles and benefit performant css-in-js
        </i>
        )
      </h2>
      <h1>
        Starter
      </h1>
      <p>
        <a
          class="btn btn-success btn-lg"
          href="/about"
        >
          <i
            class="fa fa-info"
          />
          Â  go to about
        </a>
      </p>
    </div>
  </div>
</div>
`;
